packages/core/src/contrib/Constraints.ts

```diff
@@ -229,6 +229,48 @@ export const objDict = {
   nearScalar: (c: any, goal: any) => {
     return squared(sub(constOfIf(c), constOfIf(goal)));
   },
+  /**
+   * try to make distance between a point and a segment `s1` = padding.
+   */
+  closestPtToSeg: ([, s1]: [string, any], point: VarAD[], padding: VarAD) => {
```

Comment:
Looks mostly good, though I assumed you would make this a constraint, not an objective.

The thing to note is that `equalHard` is written as a constraint (as noted in the function comment). So I would either:

- Make `closestPtToSeg` a constraint and see how well that works, OR
- Use `equalHard` squared, and keep `closestPtToSeg` as an objective. That will give it nicer numerical properties as an energy (currently it is linear and looks like this: https://www.wolframalpha.com/input/?i=f%28x%2C+y%29+%3D+max%28x%2C+y%29+-+min%28x%2C+y%29 )

Let me know how well either option works. 

Also, I'd give your function a more descriptive name, like `pointAtDistFromClosestPt`.

Reply:
I tried both of your suggestions, and it seems like keeping it as an objective is a bit faster and more reliable than making it a constraint. When I made it a constraint, every now and again it would take ~3minutes to optimize and end in a degenerated state such as this:
![Screenshot from 2021-06-28 11-38-48](https://user-images.githubusercontent.com/31522997/123687871-45f4b300-d806-11eb-8cb2-ea794b8804fb.png)
Keeping it as an objective still results in the 2-5 second time between resamples that I described in the original PR comment (as a constraint it is closer to 5 seconds on average)